config is reset on startup and ignores env

### Steps to reproduce


```yaml
---
services:
  suwayomi:
    image: ghcr.io/suwayomi/suwayomi-server:preview
    environment:
      - TZ=Europe/Paris
      - BIND_IP=0.0.0.0
      - BIND_PORT=4567
      - SOCKS_PROXY_ENABLED=false
      - DOWNLOAD_AS_CBZ=false
      - AUTH_MODE=none
      - DEBUG=false
      - MAX_LOG_FILES=1
      - MAX_LOG_FILE_SIZE=2mb
      - MAX_LOG_FOLDER_SIZE=100mb
      - WEB_UI_ENABLED=true
      - WEB_UI_FLAVOR=webUI
      - WEB_UI_CHANNEL=preview
      - WEB_UI_UPDATE_INTERVAL=23
      - AUTO_DOWNLOAD_CHAPTERS=false
      - AUTO_DOWNLOAD_EXCLUDE_UNREAD=false
      - AUTO_DOWNLOAD_NEW_CHAPTERS_LIMIT=0
      - DOWNLOAD_CONVERSIONS={}
      - EXTENSION_REPOS=["https://github.com/keiyoushi/extensions"]
      - MAX_SOURCES_IN_PARALLEL=10
      - UPDATE_EXCLUDE_UNREAD=false
      - UPDATE_EXCLUDE_STARTED=false
      - UPDATE_EXCLUDE_COMPLETED=false
      - UPDATE_INTERVAL=2
      - UPDATE_MANGA_INFO=true
      - BACKUP_TIME=05:00
      - BACKUP_INTERVAL=1
      - BACKUP_TTL=10
      - FLARESOLVERR_ENABLED=true
      - FLARESOLVERR_URL=http://byparr:8191
      - FLARESOLVERR_TIMEOUT=60
      - FLARESOLVERR_RESPONSE_AS_FALLBACK=true
      - KOREADER_SYNC_STRATEGY_FORWARD=disabled
      - KOREADER_SYNC_STRATEGY_BACKWARD=disabled
      - DATABASE_TYPE=H2
    volumes:
      - ./data:/home/suwayomi/.local/share/Tachidesk
      - ./tmp:/tmp/Tachidesk
    ports:
      - "4567:4567"
```

### Expected behavior

Suwayomi should generate a `server.conf` file matching the environment variables

### Actual behavior

Every time Suwayomi is booted up, `server.conf` is reset to the default settings.
This happens whether it's a clean install (empty volume) or a restart (pre-existing server.conf w/ values edited from webUI)

### Suwayomi-Server version

v2.1.1927

### Server operating system

Docker

### Server Desktop Environment

none

### Server JVM version

docker

### Used client name

Suwayomi-WebUI

### Client version

irrelevant

### Used web browser

irrelevant

### Client operating system

irrelevant

### Other details

bug introduced by 904157a91a4f788c2c8c018dd10b56ce67732811

```log
2025-09-14 14:38:16.982 | Suwayomi data location inside the container: /home/suwayomi/.local/share/Tachidesk
2025-09-14 14:38:17.227 | LD_PRELOAD=/opt/catch_abort.so /home/suwayomi/.local/share/Tachidesk/bin/kcef/libcef.so
2025-09-14 14:38:17.947 | 14:38:17.943 [Thread-0] INFO suwayomi.tachidesk.server.ServerSetup -- Starting Android Main Loop
2025-09-14 14:38:18.405 | 14:38:18.404 [main] INFO suwayomi.tachidesk.server.ServerSetup -- Running Suwayomi-Server v2.1.1927
2025-09-14 14:38:18.667 | 14:38:18.666 [main] INFO suwayomi.tachidesk.server.util.AppMutex -- Mutex status is clear, Resuming startup.
2025-09-14 14:38:18.752 | 14:38:18.750 [main] ERROR suwayomi.tachidesk.server.ServerSetup -- Exception while creating initial server.conf
2025-09-14 14:38:18.752 | com.typesafe.config.ConfigException$Missing: merge of /home/suwayomi/.local/share/Tachidesk/server.conf: 2-45,server-reference.conf @ jar:file:/home/suwayomi/startup/tachidesk_latest.jar!/server-reference.conf: 2-102: No configuration setting found for key 'server.koreaderSyncStrategy'
2025-09-14 14:38:18.752 | 	at com.typesafe.config.impl.SimpleConfig.findKeyOrNull(SimpleConfig.java:157)
2025-09-14 14:38:18.752 | 	at com.typesafe.config.impl.SimpleConfig.findOrNull(SimpleConfig.java:175)
2025-09-14 14:38:18.752 | 	at com.typesafe.config.impl.SimpleConfig.findOrNull(SimpleConfig.java:181)
2025-09-14 14:38:18.752 | 	at com.typesafe.config.impl.SimpleConfig.find(SimpleConfig.java:189)
2025-09-14 14:38:18.752 | 	at com.typesafe.config.impl.SimpleConfig.find(SimpleConfig.java:194)
2025-09-14 14:38:18.752 | 	at com.typesafe.config.impl.SimpleConfig.getValue(SimpleConfig.java:208)
2025-09-14 14:38:18.752 | 	at com.typesafe.config.impl.SimpleConfig.getValue(SimpleConfig.java:42)
2025-09-14 14:38:18.752 | 	at suwayomi.tachidesk.server.ServerSetupKt.applicationSetup$lambda$8(ServerSetup.kt:323)
2025-09-14 14:38:18.752 | 	at xyz.nulldev.ts.config.ConfigManager.updateUserConfig(ConfigManager.kt:172)
2025-09-14 14:38:18.752 | 	at suwayomi.tachidesk.server.ServerSetupKt.applicationSetup(ServerSetup.kt:313)
2025-09-14 14:38:18.752 | 	at suwayomi.tachidesk.MainKt.main(Main.kt:14)
2025-09-14 14:38:18.752 | 	at suwayomi.tachidesk.MainKt.main(Main.kt)
2025-09-14 14:38:18.992 | 14:38:18.992 [main] INFO suwayomi.tachidesk.server.database.DBManager -- Using H2 database version 1.4
2025-09-14 14:38:19.173 | 14:38:19.172 [main] INFO de.neonew.exposed.migrations.RunMigrations -- Running migrations on database jdbc:h2:/home/suwayomi/.local/share/Tachidesk/database
2025-09-14 14:38:19.361 | 14:38:19.360 [main] INFO de.neonew.exposed.migrations.RunMigrations -- Database version before migrations: 52
2025-09-14 14:38:19.362 | 14:38:19.362 [main] INFO de.neonew.exposed.migrations.RunMigrations -- Migrations finished successfully
2025-09-14 14:38:19.503 | 14:38:19.503 [main] INFO Migration -- Running migrations, previous version 2, target version 2
2025-09-14 14:38:19.511 | 14:38:19.510 [main] INFO Migration -- Skipping migration version 1: InitialMigration
2025-09-14 14:38:19.511 | 14:38:19.511 [main] INFO Migration -- Skipping migration version 2: FixGlobalUpdateScheduling
2025-09-14 14:38:19.527 | 14:38:19.526 [DefaultDispatcher-worker-4] INFO suwayomi.tachidesk.server.ServerSetup -- Socks Proxy changed - enabled=false address=: , username=, password=[REDACTED]
2025-09-14 14:38:19.750 | 14:38:19.749 [main] INFO suwayomi.tachidesk.server.util.WebInterfaceManager setupWebUI(flavor= WebUI, servedFlavor= WebUI) -- found webUI files - version= r2779
2025-09-14 14:38:19.750 | 14:38:19.750 [main] INFO suwayomi.tachidesk.server.util.WebInterfaceManager isLocalWebUIValid(flavor= WebUI, path= /home/suwayomi/.local/share/Tachidesk/webUI) -- Verifying WebUI files...
2025-09-14 14:38:19.913 | JCEF(38:19:912): initialized stderr logger, severity=LOGSEVERITY_DEFAULT
2025-09-14 14:38:19.913 | JCEF_I(38:19:913): CefApp: set state NEW
2025-09-14 14:38:19.935 | JCEF_I(38:19:934): CefApp: set state INITIALIZING
2025-09-14 14:38:19.935 | JCEF_V(38:19:935): Initialize CefApp on Thread[#47,CefInitialize-thread,5,main]
2025-09-14 14:38:20.027 | [0914/143820.027314:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
2025-09-14 14:38:20.054 | JCEF_V(38:20:054): CefApp: native initialization is finished.
2025-09-14 14:38:20.055 | JCEF_I(38:20:054): CefApp: set state INITIALIZED
2025-09-14 14:38:20.107 | JCEF_I(38:20:058): version: JCEF Version = 122.1.9.1056.340bc1af44ee5ffc9d212854a4dfdf2359a5b2b0
2025-09-14 14:38:20.107 | CEF Version = 122.1.9
2025-09-14 14:38:20.107 | Chromium Version = 122.0.6261.94 | settings: browser_subprocess_path=/opt/kcef/jcef/jcef_helper, windowless_rendering_enabled=true, command_line_args_disabled=false, cache_path=/home/suwayomi/.local/share/Tachidesk/cache/kcef, persist_session_cookies=false, user_agent=null, user_agent_product=null, locale=null, log_file=null, log_severity=LOGSEVERITY_DEFAULT, javascript_flags=null, resources_dir_path=/opt/kcef/jcef, locales_dir_path=/opt/kcef/jcef/locales, pack_loading_disabled=false, remote_debugging_port=0, uncaught_exception_stack_size=0, cookieable_schemes_list=null, cookieable_schemes_exclude_defaults=false, no_sandbox=true
2025-09-14 14:38:20.118 | [0914/143820.117667:WARNING:gpu_memory_buffer_support_x11.cc(49)] dri3 extension not supported.
2025-09-14 14:38:20.120 | [0914/143820.119851:WARNING:sandbox_linux.cc(418)] InitializeSandbox() called with multiple threads in process gpu-process.
2025-09-14 14:38:20.276 | 14:38:20.275 [DefaultDispatcher-worker-2] ERROR suwayomi.tachidesk.server.ServerSetup -- Failed to create/remove SystemTray due to
2025-09-14 14:38:20.276 | java.lang.ExceptionInInitializerError: null
2025-09-14 14:38:20.276 | 	at dorkbox.jna.linux.GtkEventDispatch.startGui(GtkEventDispatch.java:80)
2025-09-14 14:38:20.276 | 	at dorkbox.systemTray.SystemTray.get(SystemTray.java:651)
2025-09-14 14:38:20.276 | 	at suwayomi.tachidesk.server.util.SystemTray.create(SystemTray.kt:40)
2025-09-14 14:38:20.276 | 	at suwayomi.tachidesk.server.ServerSetupKt$applicationSetup$19.invokeSuspend(ServerSetup.kt:407)
2025-09-14 14:38:20.276 | 	at suwayomi.tachidesk.server.ServerSetupKt$applicationSetup$19.invoke(ServerSetup.kt)
2025-09-14 14:38:20.276 | 	at suwayomi.tachidesk.server.ServerSetupKt$applicationSetup$19.invoke(ServerSetup.kt)
2025-09-14 14:38:20.276 | 	at suwayomi.tachidesk.server.ServerConfig$subscribeTo$2.invokeSuspend(ServerConfig.kt:811)
2025-09-14 14:38:20.276 | 	at suwayomi.tachidesk.server.ServerConfig$subscribeTo$2.invoke(ServerConfig.kt)
2025-09-14 14:38:20.276 | 	at suwayomi.tachidesk.server.ServerConfig$subscribeTo$2.invoke(ServerConfig.kt)
2025-09-14 14:38:20.276 | 	at kotlinx.coroutines.flow.FlowKt__TransformKt$onEach$$inlined$unsafeTransform$1$2.emit(Emitters.kt:50)
2025-09-14 14:38:20.276 | 	at kotlinx.coroutines.flow.SharedFlowImpl.collect$suspendImpl(SharedFlow.kt:397)
2025-09-14 14:38:20.276 | 	at kotlinx.coroutines.flow.SharedFlowImpl$collect$1.invokeSuspend(SharedFlow.kt)
2025-09-14 14:38:20.276 | 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:34)
2025-09-14 14:38:20.276 | 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:100)
2025-09-14 14:38:20.276 | 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:586)
2025-09-14 14:38:20.276 | 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:829)
2025-09-14 14:38:20.276 | 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:717)
2025-09-14 14:38:20.276 | 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:704)
2025-09-14 14:38:20.276 | Caused by: java.lang.RuntimeException: We apologize for this, but we are unable to determine the GTK library is in use, or even if it is in use... Please create an issue for this and include your OS type and configuration.
2025-09-14 14:38:20.276 | 	at dorkbox.jna.linux.GtkLoader.<clinit>(GtkLoader.java:260)
2025-09-14 14:38:20.276 | 	... 18 common frames omitted
2025-09-14 14:38:20.607 | 14:38:20.606 [main] INFO suwayomi.tachidesk.server.util.WebInterfaceManager isLocalWebUIValid(flavor= WebUI, path= /home/suwayomi/.local/share/Tachidesk/webUI) -- Validation succeeded - md5: local= 675a16d1524ce346725b4a774948dd52; expected= 675a16d1524ce346725b4a774948dd52
2025-09-14 14:38:20.608 | 14:38:20.607 [main] INFO suwayomi.tachidesk.server.JavalinSetup -- Serving web static files for WEBUI
2025-09-14 14:38:23.131 | 14:38:23.131 [main] INFO io.javalin.Javalin -- Starting Javalin ...
2025-09-14 14:38:23.203 | 14:38:23.202 [main] INFO io.javalin.Javalin -- 
2025-09-14 14:38:23.203 |        __                  ___           _____
2025-09-14 14:38:23.203 |       / /___ __   ______ _/ (_)___      / ___/
2025-09-14 14:38:23.203 |  __  / / __ `/ | / / __ `/ / / __ \    / __ \
2025-09-14 14:38:23.203 | / /_/ / /_/ /| |/ / /_/ / / / / / /   / /_/ /
2025-09-14 14:38:23.203 | \____/\__,_/ |___/\__,_/_/_/_/ /_/    \____/
2025-09-14 14:38:23.203 | 
2025-09-14 14:38:23.203 |        https://javalin.io/documentation
2025-09-14 14:38:23.203 | 
2025-09-14 14:38:23.203 | 14:38:23.203 [main] INFO io.javalin.Javalin -- Javalin started in 140ms \o/
2025-09-14 14:38:23.220 | 14:38:23.220 [main] INFO io.javalin.Javalin -- Static file handler added: StaticFileConfig(hostedPath=/, directory=/home/suwayomi/.local/share/Tachidesk/webUI, location=EXTERNAL, precompress=false, aliasCheck=null, headers={Cache-Control=max-age=0}, mimeTypes={}, roles=[]). File system location: '/home/suwayomi/.local/share/Tachidesk/webUI'
2025-09-14 14:38:23.227 | 14:38:23.227 [main] INFO io.javalin.Javalin -- Listening on http://0.0.0.0:4567/
```

### Acknowledgements

- [x] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [x] I have written a short but informative title (ideally less than ~100 characters).
- [x] I have tried the troubleshooting guide described in [README.md](https://github.com/Suwayomi/Suwayomi-Server?tab=readme-ov-file#troubleshooting-and-support)
- [x] I have updated to the **[latest version](https://github.com/suwayomi/suwayomi-server/releases/latest)**.
- [x] I have filled out all of the requested information in this form, including specific version numbers.
- [x] I understand that **Suwayomi does not have or fix any extensions**, and I **will not receive help** for any issues related to sources or extensions.

--- a/scripts/startup_script.sh
+++ b/scripts/startup_script.sh
@@ -47,12 +47,12 @@ sed -i -r "s/server.excludeEntryWithUnreadChapters = ([0-9]+|[a-zA-Z]+)( #)?/ser
 sed -i -r "s/server.autoDownloadNewChaptersLimit = ([0-9]+|[a-zA-Z]+)( #)?/server.autoDownloadNewChaptersLimit = ${AUTO_DOWNLOAD_NEW_CHAPTERS_LIMIT:-\1} #/" /home/suwayomi/.local/share/Tachidesk/server.conf
 sed -i -r "s/server.autoDownloadIgnoreReUploads = ([0-9]+|[a-zA-Z]+)( #)?/server.autoDownloadIgnoreReUploads = ${AUTO_DOWNLOAD_IGNORE_REUPLOADS:-\1} #/" /home/suwayomi/.local/share/Tachidesk/server.conf
 if [ -n "$DOWNLOAD_CONVERSIONS" ]; then
-    perl -0777 -i -pe 's/server\.downloadConversions = ({[^#]*})/server.downloadConversions = $ENV{DOWNLOAD_CONVERSIONS}/gs' /home/suwayomi/.local/share/Tachidesk/server.conf
+    perl -0777 -i -pe 's/server\.downloadConversions = ({[^#]*?})/server.downloadConversions = $ENV{DOWNLOAD_CONVERSIONS}/gs' /home/suwayomi/.local/share/Tachidesk/server.conf
 fi
 
 # extension repos
 if [ -n "$EXTENSION_REPOS" ]; then
-    perl -0777 -i -pe 's/server\.extensionRepos = (\[.*\])/server.extensionRepos = $ENV{EXTENSION_REPOS}/gs' /home/suwayomi/.local/share/Tachidesk/server.conf
+    perl -0777 -i -pe 's/server\.extensionRepos = (\[.*?\])/server.extensionRepos = $ENV{EXTENSION_REPOS}/gs' /home/suwayomi/.local/share/Tachidesk/server.conf
 fi
 
 # requests

